sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1223,37 +1272,36 @@ def task_worker(self):
                         has_update = self.is_cmis_application_update_required(api, host_lanes, host_speed)
                         if not has_update:
                             # No application updates
-                            self.log_notice("{}: READY".format(lport))
+                            self.log_notice("{}: no CMIS application update required...READY".format(lport))
                             self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_READY
                             continue
 
+                        self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_DP_DEINIT
+                    elif state == self.CMIS_STATE_DP_DEINIT:
                         # D.2.2 Software Deinitialization
                         api.set_datapath_deinit(host_lanes)
-                        api.set_lpmode(True)
-                        if not self.test_module_state(api, ['ModuleReady', 'ModuleLowPwr']):
-                            self.log_notice("{}: unable to enter low-power mode".format(lport))
-                            self.port_dict[lport]['cmis_retries'] = retries + 1
-                            continue
 
                         # D.1.3 Software Configuration and Initialization
                         if not api.tx_disable_channel(host_lanes, True):
                             self.log_notice("{}: unable to turn off tx power".format(lport))
                             self.port_dict[lport]['cmis_retries'] = retries + 1
                             continue
-                        api.set_lpmode(False)
 
-                        # TODO: Use fine grained time when the CMIS memory map is available
+                        # TODO: Make sure this doesn't impact other datapaths
+                        api.set_lpmode(False)
```

Comment:
if I understand correctly, the previous  `api.set_lpmode(True)` has been removed, should this line also need to be removed?

Reply:
This is a precondition for Datapath Activation and hence required.